### PR TITLE
pytorch 2.0.0 torchvision 0.15.1

### PR DESCRIPTION
**Pull request by chenrui333**
Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by carlocab** [comment]
```
    Running command Getting requirements to build wheel
    No sympy found
    Traceback (most recent call last):
      File "/opt/homebrew/Cellar/torchvision/0.15.1/libexec/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
        main()
      File "/opt/homebrew/Cellar/torchvision/0.15.1/libexec/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/homebrew/Cellar/torchvision/0.15.1/libexec/lib/python3.11/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 118, in get_requires_for_build_wheel
        return hook(config_settings)
               ^^^^^^^^^^^^^^^^^^^^^
      File "/private/tmp/pip-build-env-o8imt_hb/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in get_requires_for_build_wheel
        return self._get_build_requires(config_settings, requirements=['wheel'])
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/private/tmp/pip-build-env-o8imt_hb/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 320, in _get_build_requires
        self.run_setup()
      File "/private/tmp/pip-build-env-o8imt_hb/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 485, in run_setup
        self).run_setup(setup_script=setup_script)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "/private/tmp/pip-build-env-o8imt_hb/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 335, in run_setup
        exec(code, locals())
      File "<string>", line 9, in <module>
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/__init__.py", line 1465, in <module>
        from . import _meta_registrations
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_meta_registrations.py", line 7, in <module>
        from torch._decomp import _add_op_to_registry, global_decomposition_table, meta_table
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_decomp/__init__.py", line 169, in <module>
        import torch._decomp.decompositions
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_decomp/decompositions.py", line 10, in <module>
        import torch._prims as prims
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_prims/__init__.py", line 33, in <module>
        from torch._subclasses.fake_tensor import FakeTensor, FakeTensorMode
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_subclasses/__init__.py", line 3, in <module>
        from torch._subclasses.fake_tensor import (
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_subclasses/fake_tensor.py", line 13, in <module>
        from torch._guards import Source
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_guards.py", line 78, in <module>
        class ShapeGuard(NamedTuple):
      File "/opt/homebrew/opt/pytorch/lib/python3.11/site-packages/torch/_guards.py", line 79, in ShapeGuard
        expr: sympy.Expr
              ^^^^^
    NameError: name 'sympy' is not defined
    error: subprocess-exited-with-error
    
    × Getting requirements to build wheel did not run successfully.
    │ exit code: 1
    ╰─> See above for output.
```
I guess this needs a sympy resource?

**Comment by chenrui333** [comment]
kick a run.

**Comment by carlocab** [comment]
6ab9c2cb989f3a352982791b0263e61216d4dd3e modifies 2 files. It needs to be split into two separate commits.

**Comment by singingwolfboy** [comment]
> [6ab9c2c](https://github.com/Homebrew/homebrew-core/commit/6ab9c2cb989f3a352982791b0263e61216d4dd3e) modifies 2 files. It needs to be split into two separate commits.

Fixed! Now hopefully the build is fixed, too... 🤞

**Comment by chenrui333** [comment]
`Failed to execute: /opt/homebrew/Cellar/torchvision/0.15.1/libexec/bin/pip`

**Comment by singingwolfboy** [comment]
> `Failed to execute: /opt/homebrew/Cellar/torchvision/0.15.1/libexec/bin/pip`

Yeah, I see that, but I don't understand what's causing it. Could it be a temporary, transient issue? I can't replicate it locally.

**Comment by chenrui333** [comment]
let's do another run then.

File: Formula/torchvision.rb
Comment (carlocab) [inline Formula/torchvision.rb:1]: Yea, I think this should help too 👍

File: Formula/torchvision.rb
Comment (carlocab) [inline Formula/torchvision.rb:86]: Try doing
```ruby
ENV.append_path "PYTHONPATH",pytorch/site_packages
```
before calling `virtual_env_install_with_resources`. _Then_ you can write the `.pth` file.

Comment (singingwolfboy) [reply Formula/torchvision.rb:86]: Ugh, still not working. Any other suggestions? You're welcome to just push commits to this branch, as well.

File: Formula/torchvision.rb
Comment (singingwolfboy) [inline Formula/torchvision.rb:84]: This depends on https://github.com/Homebrew/brew/pull/15121

**Comment by singingwolfboy** [comment]
All the builds are finally passing! @carlocab @chenrui333 could one or both of you please review this code?

Also, what's the process for merging this? Any idea when https://github.com/Homebrew/brew/pull/15121 might land in a release tag?

File: Formula/openai-whisper.rb
Comment (carlocab) [inline Formula/openai-whisper.rb:78]: You could probably write all these paths into a single `.pth` file.

File: Formula/pytorch.rb
Comment (carlocab) [inline Formula/pytorch.rb:41]: You can put this outside of the `on_macos` block.

**Comment by chenrui333** [comment]
Thanks @singingwolfboy!! 💯

**Comment by dawidd6** [comment]
Big Sur bottles were dropped?

**Review by carlocab — APPROVED** [review]

**Comment by carlocab** [comment]
Not stuck; just waiting for an approval. 😄

**Comment by dawidd6** [comment]
lol, you're right...

**Pull request merged**
